src/components/app/routes/data/utils.js

```diff
@@ -240,23 +241,21 @@ export async function ensureAuthenticatedUser(requestUrl, params) {
  * TODO
  * @param {*} enterpriseSlug
  * @param {*} activeEnterpriseCustomer
+ * @param {*} staffEnterpriseCustomer
  * @param {*} allLinkedEnterpriseCustomerUsers
  * @param {*} requestUrl
- * @param {*} queryClient
- * @param {*} updateUserActiveEnterprise
  * @returns
  */
 export async function ensureActiveEnterpriseCustomerUser({
   enterpriseSlug,
   activeEnterpriseCustomer,
+  staffEnterpriseCustomer,
   allLinkedEnterpriseCustomerUsers,
   requestUrl,
-  queryClient,
-  username,
-  // updateUserActiveEnterprise,
 }) {
-  // If the enterprise slug in the URL matches the active enterprise customer's slug, return early.
-  if (enterpriseSlug === activeEnterpriseCustomer.slug) {
+  // If the enterprise slug in the URL matches the active enterprise customer user's slug OR the
+  // slug returned by the staff metadata, return early.
+  if (!activeEnterpriseCustomer || activeEnterpriseCustomer.slug === enterpriseSlug) {
```

Comment:
nit: "the slug returned by the staff metadata" doesn't seem to line up with `!activeEnterpriseCustomer` here, but maybe I'm missing something?

Reply:
Ah, great catch. This is an outdated comment; logic was changed since it was written. Updated!